test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusJunitCallbacks.java

```diff
@@ -23,7 +23,9 @@ static void invokeJunitBefores(String className, Object testInstance)
             collectCallbacks(testInstance.getClass(), befores, (Class<? extends Annotation>) testInstance.getClass()
                     .getClassLoader().loadClass(className));
             for (Method before : befores) {
-                before.invoke(testInstance);
+                if (before.canAccess(testInstance) && before.getParameters().length == 0) {
```

Comment:
I wonder if we shouldn't log why Before and After methods were not invoked when this returns false

Reply:
Yes, this is not ideal, but it is just an easy fix to run the JAX-RS TCK. Annotated methods can declare parameters resolvable by JUnit `ParameterResolvers.`. In such cases, we will fail. This fix will ignore such methods, but we need a more permanent solution.